### ISSUE TITLE
Should display names of object properties in view

### ISSUE THREAD

**Issue by heapifyman** [protegeproject/ontograf#10]
Names of object properties (or whatever you call the arrows that are connecting classes) should be displayed.
Currently only the class names are shown which makes the tool completely useless for getting a visual overview of an ontology.

**Comment by csnyulas** [comment]
Different types of object properties are represented with different 
types of arrows in OntoGraf. You can turn on the legend for the arrows 
by clicking on the "Arc Types" icon (showing a blue arrow and a 
checkbox) in the task bar. This will give you a very nice overview of 
what everything means on your graph, and more, you are even able to 
control which type of relations you want to see in your graph.

Cheers,
Csongor

**Comment by heapifyman** [comment]
I know that "feature".

But once you have a somewhat complex ontology and more than say 20 object properties, this list becomes

- a mess of colors that only differ very slightly - maybe I'm a bit color blind but I cannot immediately grasp the difference between two different shades of purple and compare them two two thin lines in another part of the screen, so I basically have to guess which line means what
- too long to fit on the screen - so I have to scroll to see what a specific line in the diagram is supposed to mean

In short, I think this "solution" offers very poor usability.

**Comment by heapifyman** [comment]
How do I re-open this issue?

**Comment by csnyulas** [comment]
I reopened the issue for you, although we are not planning to work on this in the near future. Maybe somebody will be interested in contributing.

**Comment by csnyulas** [comment]
This feature was requested by another user on our mailing list:
http://protege-project.136.n4.nabble.com/ObjectProperty-Visualization-in-Protege-td4667446.html

**Comment by csnyulas** [comment]
Regarding the disadvantages of the current solution listed by @heapifyman, just wanted to note that if you have a complex ontology with a lot of relations, and you try to visualize a set of classes having a large number of relationships among them, adding all those labels to all the arcs will make an even bigger "mess" on the graph, making it potentially totally unreadable. Just a thought to consider.

**Comment by lingvisa** [comment]
Have this requested feature is definitely helpful in many senses. To visualize a subgraph  and take a snapshot to a PPT slide, it's extremely nice to be able to display object property, which is an essential part of most ontologies.

**Comment by fahedAlkhabbas** [comment]
I am really looking for this feature!

**Comment by eric-jahn** [comment]
It would be great to be able to see the data properties that are in the domain of a given class.  It would be good to see the range of that property as well, next to the property name, within the class.
